core/distributed/matrix.cpp

```diff
@@ -89,6 +88,62 @@ Matrix<ValueType, LocalIndexType, GlobalIndexType>::Matrix(
     local_mtx_ = local_linop;
 }
 
+template <typename ValueType, typename LocalIndexType, typename GlobalIndexType>
+Matrix<ValueType, LocalIndexType, GlobalIndexType>::Matrix(
```

Comment:
This introduces a new way to create and fill distributed matrices compared to the read_distributed approach. I find that a bit confusing because we don't really communicate why in one situation on use  `Matrix::create(..)` followed by `Matrix::read_distributed()` vs `Matrix::create(..)` and passing the local and non-local matrix directly.